app/src/features/summary/list-data/project/ProjectsListContainer.tsx

```diff
@@ -0,0 +1,277 @@
+import Box from '@mui/material/Box';
+import Collapse from '@mui/material/Collapse';
+import grey from '@mui/material/colors/grey';
+import Divider from '@mui/material/Divider';
+import Link from '@mui/material/Link';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { GridColDef, GridPaginationModel, GridSortDirection, GridSortModel } from '@mui/x-data-grid';
+import ColouredRectangleChip from 'components/chips/ColouredRectangleChip';
+import { StyledDataGrid } from 'components/data-grid/StyledDataGrid';
+import { SystemRoleGuard } from 'components/security/Guards';
+import { getNrmRegionColour, NRM_REGION_APPENDED_TEXT } from 'constants/regions';
+import { SYSTEM_ROLE } from 'constants/roles';
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import { useCodesContext, useTaxonomyContext } from 'hooks/useContext';
+import useDataLoader from 'hooks/useDataLoader';
+import { useDeepCompareEffect } from 'hooks/useDeepCompareEffect';
+import { useSearchParams } from 'hooks/useSearchParams';
+import { IProjectsListItemData } from 'interfaces/useProjectApi.interface';
+import { useEffect, useMemo, useState } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+import { ApiPaginationRequestOptions, StringValues } from 'types/misc';
+import { firstOrNull, getCodesName } from 'utils/Utils';
+import ProjectsListFilterForm, {
+  IProjectAdvancedFilters,
+  ProjectAdvancedFiltersInitialValues
+} from './ProjectsListFilterForm';
+
+// Supported URL parameters
+type ProjectDataTableURLParams = {
+  // filter
+  p_keyword?: string;
+  p_itis_tsn?: number;
+  p_system_user_id?: string;
+  // pagination
+  p_page?: string;
+  p_limit?: string;
+  p_sort?: string;
+  p_order?: 'asc' | 'desc';
+};
+
+const pageSizeOptions = [10, 25, 50];
+
+interface IProjectsListContainerProps {
+  showSearch: boolean;
+}
+
+// Default pagination parameters
+const ApiPaginationRequestOptionsInitialValues: Required<ApiPaginationRequestOptions> = {
+  page: 0,
+  limit: 10,
+  sort: 'project_id',
+  order: 'desc'
+};
+
+/**
+ * Displays a list of projects.
+ *
+ * @return {*}
+ */
+const ProjectsListContainer = (props: IProjectsListContainerProps) => {
+  const { showSearch } = props;
+
+  const biohubApi = useBiohubApi();
+  const codesContext = useCodesContext();
+  const taxonomyContext = useTaxonomyContext();
+
+  const { searchParams, setSearchParams } = useSearchParams<StringValues<ProjectDataTableURLParams>>();
+
+  useEffect(() => {
+    codesContext.codesDataLoader.load();
+  }, [codesContext.codesDataLoader]);
+
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({
```

Comment:
I think some comments here around the setStates would be helpful and maybe why we prepend 'p_' on the properties. (I think it was something to do with multiple tables being rendered).

Reply:
added!